tests/metrics/test_divergence.py

```diff
@@ -143,6 +138,54 @@ def test_divergence_single(self):
 
         pd.testing.assert_frame_equal(result, expected_df)
 
+    def test_divergence_duplicate(self):
+        """p3.data.divergence.duplicate"""
+        data = {
+            "problem": ["test"] * 2,
+            "platform": ["A", "B"],
+            "application": ["latest"] * 2,
+            "coverage_key": ["source1", "source2"],
+        }
+        df = pd.DataFrame(data)
+
+        source1_json_string = json.dumps(
+            [
+                {
+                    "file": "foo.cpp",
+                    "id": "0",
+                    "lines": [[0, 9]],
+                }
+            ]
+        )
+
+        source2_json_string = json.dumps(
+            [
+                {
+                    "file": "foo.cpp",
+                    "id": "1",
```

Comment:
You might want to add inline comments for this and the other, just to draw attention to it.

Reply:
I put some explanatory comments before the definitions in 99730528835e65934096fc8a123d1cc1d22e235c.